### ISSUE TITLE
HTTP probe to google.com or cloudprober.org fails when disable_http2 is set to true

### ISSUE THREAD

**Issue by edbertlinardi** [cloudprober/cloudprober#872]
**Describe the bug**
When `disable_http2` option is set to true, HTTP probe to websites like google.com or cloudprober.org is failing with 
```
time=2024-10-02T13:39:38.208-07:00 level=WARN source=probes/http/http.go:342 msg="Get \"https://google.com\": net/http: HTTP/1.x transport connection broken: malformed HTTP response \"\\x00\\x00\\x12\\x04\\x00\\x00\\x00\\x00\\x00\\x00\\x03\\x00\\x00\\x00d\\x00\\x04\\x00\\x10\\x00\\x00\\x00\\x06\\x00\\x01\\x00\\x00\\x00\\x00\\x04\\b\\x00\\x00\\x00\\x00\\x00\\x00\\x0f\\x00\\x01\\x00\\x00\\x1e\\a\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x01http2_handshake_failed\"" system=cloudprober probe=google target=google.com url=https://google.com
time=2024-10-02T13:39:40.223-07:00 level=WARN source=probes/http/http.go:342 msg="Get \"https://cloudprober.org\": net/http: HTTP/1.x transport connection broken: malformed HTTP response \"\\x00\\x00\\x12\\x04\\x00\\x00\\x00\\x00\\x00\\x00\\x03\\x00\\x00\\x00d\\x00\\x04\\x00\\x01\\x00\\x00\\x00\\x05\\x00\\xff\\xff\\xff\\x00\\x00\\x04\\b\\x00\\x00\\x00\\x00\\x00\\x7f\\xff\\x00\\x00\\x00\\x00\\b\\a\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x01\"" system=cloudprober probe=cloudprober target=cloudprober.org url=https://cloudprober.org
```
**Cloudprober Version**
v0.13.7

**To Reproduce**
Run cloudprober with the following config
```
probe {
  name: "google"
  type: HTTP
  targets {
    host_names: "google.com"
  }
  interval_msec: 5000  # 5s
  timeout_msec: 1000   # 1s

  http_probe {
    protocol: HTTPS
    disable_http2: true
  }
}

probe {
  name: "cloudprober"
  type: HTTP
  targets {
    host_names: "cloudprober.org"
  }
  interval_msec: 5000  # 5s
  timeout_msec: 1000   # 1s

  http_probe {
    protocol: HTTPS
    disable_http2: true
  }
}
```

**Additional context**

**Comment by manugarg** [comment]
Thanks for filing this @edbertlinardi. As we discussed offline, problem is that cloning transport before setting TLSNextProto to an empty map[1], makes the Go's recommendation for how to disable HTTP2 completely useless, as first clone configures HTTP2 in the transport[2]. I think Go should either change the recommendation or make TLSNextProto being an empty map always work. I think https://github.com/golang/go/issues/39302 covers the Go issue.
[1] - https://github.com/cloudprober/cloudprober/blob/f031b8f8e3f8bfc0f0e5b8510b5395b51fcfda58/probes/http/http.go#L107
[2] - https://github.com/golang/go/blob/630d4fb600ae309476c87a4ca0e29b60425bc0c1/src/net/http/transport.go#L380

**Comment by manugarg** [comment]
Fixed by #873

**Comment by cbroglie** [comment]
I stumbled across this as well in v0.13.7, and had written a small test based on https://pkg.go.dev/net/http/httptest#example-Server-HTTP2 to demonstrate the problem before finding this issue. Sharing it in case it's useful, as it does confirm the fix in #873:
```go
func TestDisableHTTP2(t *testing.T) {
	ts := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		fmt.Fprintf(w, "Hello, %s", r.Proto)
	}))
	ts.EnableHTTP2 = true
	ts.StartTLS()
	defer ts.Close()

	p := &Probe{
		opts: options.DefaultOptions(),
		c: &configpb.ProbeConf{
			DisableHttp2:          proto.Bool(true),
			DisableCertValidation: proto.Bool(true),
		},
	}

	transport, err := p.getTransport()
	if err != nil {
		t.Errorf("Unexpected error: %v", err)
	}

	client := ts.Client()
	client.Transport = transport

	res, err := client.Get(ts.URL)
	if err != nil {
		t.Fatal(err)
	}
	defer res.Body.Close()

	greeting, err := io.ReadAll(res.Body)
	if err != nil {
		t.Fatal(err)
	}
	assert.Equal(t, "Hello, HTTP/1.1", string(greeting))
}
```

**Comment by manugarg** [comment]
Thanks so much @cbroglie for this test. Added it: #992.
